packages/web-components/fast-foundation/src/virtual-list/virtual-list.ts

```diff
@@ -0,0 +1,885 @@
+import {
+    attr,
+    DOM,
+    Notifier,
+    nullableNumberConverter,
+    Observable,
+    observable,
+    RepeatBehavior,
+    RepeatDirective,
+    Splice,
+    ViewTemplate,
+} from "@microsoft/fast-element";
+import { eventResize, eventScroll, Orientation } from "@microsoft/fast-web-utilities";
+import { FoundationElement } from "../foundation-element";
+import { IntersectionService } from "../utilities/intersection-service";
+import { IdleCallbackQueue } from "../utilities/idle-callback-queue";
+import type { ResizeObserverClassDefinition } from "../utilities/resize-observer";
+/**
+ * Defines when the component updates its position automatically.
+ *
+ * @public
+ */
+export type VirtualListAutoUpdateMode = "manual" | "viewport-resize" | "auto";
+
+/**
+ * Used to describe the position of an element within the list
+ *
+ * @public
+ */
+export interface SpanMap {
+    start: number;
+    end: number;
+    span: number;
+}
+
+/**
+ *  The VirtualList class
+ *
+ * @public
+ */
+export class VirtualList extends FoundationElement {
+    /**
+     *  Whether or not the display should virtualize
+     *
+     * @public
+     */
+    @attr({ attribute: "virtualize", mode: "boolean" })
+    public virtualize: boolean = true;
+    private virtualizeChanged(): void {
+        if (this.$fastController.isConnected) {
+            this.reset();
+        }
+    }
+
+    /**
+     * The HTML ID of the viewport element
+     *
+     * @public
+     * @remarks
+     * HTML Attribute: anchor
+     */
+    @attr({ attribute: "viewport" })
+    public viewport: string = "";
+    private viewportChanged(): void {
+        if (this.$fastController.isConnected) {
+            this.viewportElement = this.getViewport();
+            this.updateDimensions();
+        }
+    }
+
+    /**
+     * The span in pixels of each item along the virtualization axis
```

Comment:
"Span" makes me think of a grid. I typically refer to px generally as "size" or "measurement". I get it's not "width" or "height" because the orientation changes.

Reply:
I have a soft preference for "span" over "size", I guess "spanMap" just sounds better to me than "sizeMap".  Happy to go with the consensus.   Naming things is hard.

Reply:
went with 'size'